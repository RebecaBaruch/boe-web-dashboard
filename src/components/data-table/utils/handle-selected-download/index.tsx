import { pdf } from '@react-pdf/renderer';
import {
  PDFDocumentGenerator,
  PDFDocumentProps,
} from '../../../../../src/utils/pdf-document-generator';

const downloadPDF = async (
  record: PDFDocumentProps,
  fileName: string = 'diagnosis_report.pdf',
) => {
  const pdfDocument = PDFDocumentGenerator(record) as React.ReactElement;

  const blob = await pdf(pdfDocument).toBlob();

  const link = document.createElement('a');
  link.href = URL.createObjectURL(blob);
  link.download = fileName;
  link.click();
};

export default downloadPDF;
