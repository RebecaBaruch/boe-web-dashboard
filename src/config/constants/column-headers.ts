import { Column } from 'components/data-table/types';

export const accountsColumns: Column[] = [
  { header: 'Id', accessor: 'id' },
  { header: 'Foto', accessor: 'picture' },
  { header: 'Nome', accessor: 'name' },
  { header: 'E-mail', accessor: 'email' },
  { header: 'N° de registros', accessor: 'registers' },
  { header: 'Data de ingresso', accessor: 'date' },
  { header: 'Ação', accessor: 'action' },
];
