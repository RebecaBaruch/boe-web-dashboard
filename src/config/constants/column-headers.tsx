import React from 'react';
import {
  Column,
  TableAnimalsData,
  TableDiagnosisData,
} from 'components/data-table/types';
import {
  AnimalIdContainer,
  AnimalPhoto,
  AuthorContainer,
  AuthorPhoto,
  ResultBadge,
  RiskBadge,
  RiskIndicator,
  StatusBadge,
} from 'components/diagnosis-components/diagnosis-components.components';

export const accountsColumns: Column[] = [
  { header: 'Id', accessor: 'id' },
  { header: 'Foto', accessor: 'picture' },
  { header: 'Nome', accessor: 'name' },
  { header: 'E-mail', accessor: 'email' },
  { header: 'N° de registros', accessor: 'registers' },
  { header: 'Data de ingresso', accessor: 'date' },
];

export const diagnosisListColumns: Column[] = [
  {
    header: '',
    accessor: 'riskLabel',
    render: (item: TableDiagnosisData): React.ReactNode => (
      <RiskIndicator risk={item.risk} />
    ),
  },
  {
    header: 'Id do animal',
    accessor: 'animalId',
    render: (item: TableDiagnosisData) => (
      <AnimalIdContainer>
        <AnimalPhoto src={item.animalPhoto} alt="Animal" />
        <span>{item.animalId}</span>
      </AnimalIdContainer>
    ),
  },
  {
    header: 'Resultado',
    accessor: 'result',
    render: (item: TableDiagnosisData) => (
      <ResultBadge result={item.result}>{item.result}</ResultBadge>
    ),
  },
  {
    header: 'Risco (em %)',
    accessor: 'risk',
    render: (item: TableDiagnosisData) => (
      <RiskBadge risk={item.risk}>{item.risk}%</RiskBadge>
    ),
  },
  {
    header: 'Autor da análise',
    accessor: 'author',
    render: (item: TableDiagnosisData) => (
      <AuthorContainer>
        <AuthorPhoto src={item.authorPhoto} alt="Author" />
        <span>{item.authorName}</span>
      </AuthorContainer>
    ),
  },
  {
    header: 'Data',
    accessor: 'date',
    render: (item: TableDiagnosisData) => <span>{item.date}</span>,
  },
];

export const animalsListColumns: Column[] = [
  {
    header: '',
    accessor: 'riskLabel',
    render: (item: TableAnimalsData): React.ReactNode => (
      <RiskIndicator risk={item.risk} />
    ),
  },
  {
    header: 'Id do animal',
    accessor: 'animalId',
    render: (item: TableAnimalsData) => (
      <AnimalIdContainer>
        <AnimalPhoto src={item.animalPhoto} alt="Animal" />
        <span>{item.animalId}</span>
      </AnimalIdContainer>
    ),
  },
  {
    header: 'Tipo de infecção',
    accessor: 'result',
    render: (item: TableAnimalsData) => (
      <ResultBadge result={item.result}>{item.result}</ResultBadge>
    ),
  },
  {
    header: 'Risco (em %)',
    accessor: 'risk',
    render: (item: TableAnimalsData) => (
      <RiskBadge risk={item.risk}>{item.risk}%</RiskBadge>
    ),
  },
  {
    header: 'Data',
    accessor: 'date',
    render: (item: TableAnimalsData) => <span>{item.date}</span>,
  },
  {
    header: 'Status de tratamento',
    accessor: 'status',
    render: (item: TableAnimalsData) => (
      <StatusBadge status={item.status ?? 'Indefinido'}>
        {item.status ?? 'Indefinido'}
      </StatusBadge>
    ),
  },
];
