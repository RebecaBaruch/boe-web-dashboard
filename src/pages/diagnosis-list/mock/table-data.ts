import {
  TableAnimalsData,
  TableDiagnosisData,
} from 'components/data-table/types';

export const diagnosisMockData: TableDiagnosisData[] = Array.from(
  { length: 20 },
  (_, index) => ({
    id: `AU${String(1000 + index).padStart(4, '0')}`, // Altere para 'id' ao invés de 'animalId'
    riskLabel: ['Alto', 'Médio', 'Baixo'][Math.floor(Math.random() * 3)],
    animalId: `AU${String(1000 + index).padStart(4, '0')}`, // Deixe o 'animalId' também para manter consistência
    animalPhoto: `https://placekitten.com/50/50?image=${index}`,
    authorName: ['Nelson Oliveira', 'Fabrício Neves'][index % 2],
    authorPhoto: `https://randomuser.me/api/portraits/men/${index}.jpg`,
    date: new Date(2024, 8, 17 + (index % 7)).toLocaleDateString('pt-BR'),
    diagnosisDate: new Date(2024, 8, 17 + (index % 7)).toLocaleDateString('pt-BR'),
    result: ['Berne', 'Dermatite', 'Dermatofitose', 'Dermatofilose'][
      Math.floor(Math.random() * 4)
    ],
    risk: Math.floor(Math.random() * 101),
  }),
);

export const animalsMockData: TableAnimalsData[] = Array.from(
  { length: 10 },
  (_, index) => ({
    id: `AU${String(1000 + index).padStart(4, '0')}`, // Altere para 'id' ao invés de 'animalId'
    riskLabel: ['Alto', 'Médio', 'Baixo'][Math.floor(Math.random() * 3)],
    animalId: `AU${String(1000 + index).padStart(4, '0')}`, // Deixe o 'animalId' também para manter consistência
    animalPhoto: `https://placekitten.com/50/50?image=${index}`,
    date: new Date(2024, 8, 17 + (index % 7)).toLocaleDateString('pt-BR'),
    result: ['Berne', 'Dermatite', 'Dermatofitose', 'Dermatofilose'][
      Math.floor(Math.random() * 4)
    ],
    risk: Math.floor(Math.random() * 101),
    status: ['Em andamento', 'Concluído', 'Sem tratamento'][
      Math.floor(Math.random() * 3)
    ],
  }),
);
