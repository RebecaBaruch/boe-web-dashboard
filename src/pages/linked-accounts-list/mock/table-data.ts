export const dataMock = [
  {
    id: '1',
    picture: 'link_da_foto_1',
    name: 'Claudio Buarque',
    email: 'claudio.buarque@gmail.com',
    registers: '02',
    date: '10/01/2024',
    action: 'Ação 1',
  },
  {
    id: '2',
    picture: 'link_da_foto_2',
    name: 'Fabrício Neves',
    email: 'fabricio.neves@gmail.com',
    registers: '12',
    date: '10/01/2024',
    action: 'Ação 2',
  },
  {
    id: '3',
    picture: 'link_da_foto_3',
    name: 'José Henrique',
    email: 'jose.henrique@gmail.com',
    registers: '03',
    date: '02/02/2024',
    action: 'Ação 3',
  },
  {
    id: '4',
    picture: 'link_da_foto_4',
    name: 'Luana Buarque',
    email: 'luana.buarque@gmail.com',
    registers: '02',
    date: '05/03/2024',
    action: 'Ação 4',
  },
  {
    id: '5',
    picture: 'link_da_foto_5',
    name: 'Augusto Buarque',
    email: 'augusto.buarque@gmail.com',
    registers: '04',
    date: '02/02/2024',
    action: 'Ação 5',
  },
  {
    id: '6',
    picture: 'link_da_foto_6',
    name: 'Nelson Oliveira',
    email: 'nelson.oliveira@gmail.com',
    registers: '07',
    date: '10/01/2024',
    action: 'Ação 6',
  },
  {
    id: '7',
    picture: 'link_da_foto_7',
    name: 'Mariana Silva',
    email: 'mariana.silva@gmail.com',
    registers: '01',
    date: '11/01/2024',
    action: 'Ação 7',
  },
  {
    id: '8',
    picture: 'link_da_foto_8',
    name: 'Roberto Carlos',
    email: 'roberto.carlos@gmail.com',
    registers: '05',
    date: '15/01/2024',
    action: 'Ação 8',
  },
  {
    id: '9',
    picture: 'link_da_foto_9',
    name: 'Tatiane Lopes',
    email: 'tatiane.lopes@gmail.com',
    registers: '06',
    date: '20/01/2024',
    action: 'Ação 9',
  },
  {
    id: '10',
    picture: 'link_da_foto_10',
    name: 'Ricardo Mendes',
    email: 'ricardo.mendes@gmail.com',
    registers: '10',
    date: '25/01/2024',
    action: 'Ação 10',
  },
  {
    id: '11',
    picture: 'link_da_foto_11',
    name: 'Aline Pires',
    email: 'aline.pires@gmail.com',
    registers: '03',
    date: '30/01/2024',
    action: 'Ação 11',
  },
  {
    id: '12',
    picture: 'link_da_foto_12',
    name: 'Samuel Gomes',
    email: 'samuel.gomes@gmail.com',
    registers: '04',
    date: '02/02/2024',
    action: 'Ação 12',
  },
  {
    id: '13',
    picture: 'link_da_foto_13',
    name: 'Juliana Santos',
    email: 'juliana.santos@gmail.com',
    registers: '09',
    date: '05/02/2024',
    action: 'Ação 13',
  },
  {
    id: '14',
    picture: 'link_da_foto_14',
    name: 'Fernanda Lima',
    email: 'fernanda.lima@gmail.com',
    registers: '11',
    date: '10/02/2024',
    action: 'Ação 14',
  },
  {
    id: '15',
    picture: 'link_da_foto_15',
    name: 'Gustavo Rocha',
    email: 'gustavo.rocha@gmail.com',
    registers: '08',
    date: '15/02/2024',
    action: 'Ação 15',
  },
  {
    id: '16',
    picture: 'link_da_foto_16',
    name: 'Bruna Almeida',
    email: 'bruna.almeida@gmail.com',
    registers: '02',
    date: '20/02/2024',
    action: 'Ação 16',
  },
  {
    id: '17',
    picture: 'link_da_foto_17',
    name: 'Lucas Ferreira',
    email: 'lucas.ferreira@gmail.com',
    registers: '05',
    date: '25/02/2024',
    action: 'Ação 17',
  },
  {
    id: '18',
    picture: 'link_da_foto_18',
    name: 'Larissa Costa',
    email: 'larissa.costa@gmail.com',
    registers: '06',
    date: '01/03/2024',
    action: 'Ação 18',
  },
  {
    id: '19',
    picture: 'link_da_foto_19',
    name: 'Vitor Hugo',
    email: 'vitor.hugo@gmail.com',
    registers: '12',
    date: '05/03/2024',
    action: 'Ação 19',
  },
  {
    id: '20',
    picture: 'link_da_foto_20',
    name: 'Sofia Martins',
    email: 'sofia.martins@gmail.com',
    registers: '07',
    date: '10/03/2024',
    action: 'Ação 20',
  },
];
