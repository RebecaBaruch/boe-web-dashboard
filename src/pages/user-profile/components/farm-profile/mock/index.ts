export const mockFarmData = {
  propertyName: 'Fazenda São João',
  farmPhone: '(13) 12345-6789',
  cep: '11900-000',
  cnpj: '12.345.678/0001-99',
  location: 'Itaquá, SP',
  address: 'Rua das Palmeiras, 123, Bairro Verde',
};
