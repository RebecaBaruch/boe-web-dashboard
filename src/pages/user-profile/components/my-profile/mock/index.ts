export const mockMyProfileData = {
  firstName: 'Claudio',
  lastName: 'Buarque',
  email: 'claudio.buarque@gmail.com',
  phone: '(11) 12345-6789',
  location: 'Registro, SP, Brasil',
};
