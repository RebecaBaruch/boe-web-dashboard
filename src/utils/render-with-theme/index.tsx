import React from 'react';
import { ThemeContextProvider } from '@telefonica/mistica';
import { render, RenderResult } from '@testing-library/react';
import { theme } from '../../config/theme';

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeContextProvider theme={theme}>{children}</ThemeContextProvider>,
  );
};
