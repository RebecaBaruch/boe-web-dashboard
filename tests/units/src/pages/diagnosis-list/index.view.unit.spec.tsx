import React from 'react';
import { screen } from '@testing-library/react';
import DiagnosisList from '../../../../../src/pages/diagnosis-list/view/diagnosis-list.view';
import { renderWithTheme } from '../../../../../src/utils/render-with-theme';
import { diagnosisMockData } from '../../../../../src/pages/diagnosis-list/mock/table-data';
import { diagnosisListColumns } from '../../../../../src/config/constants/column-headers';

jest.mock('../../../../../src/services/api/boe-api-v2', () => ({
  getAnalysisHistory: jest.fn(() => Promise.resolve(diagnosisMockData)),
}));

describe('DiagnosisList', () => {
  it('should render DiagnosisList with expected headers', async () => {
    renderWithTheme(
      <DiagnosisList
        diagnosisData={diagnosisMockData}
        columnsHeader={diagnosisListColumns}
        currentPage="1"
        lastPage="3"
        prevPage={jest.fn()}
        nextPage={jest.fn()}
        toggleSelectRow={jest.fn()}
        toggleSelectAll={jest.fn()}
        onSelectMode={jest.fn()}
        activeTab={'anality'}
        setActiveTab={jest.fn()}
      />,
    );

    expect(screen.getByText('Id do animal')).toBeInTheDocument();
    expect(screen.getByText('Resultado')).toBeInTheDocument();
    expect(screen.getByText('Risco (em %)')).toBeInTheDocument();
    expect(screen.getByText('Autor da análise')).toBeInTheDocument();
    expect(screen.getByText('Data')).toBeInTheDocument();
    expect(screen.getByText('Ação')).toBeInTheDocument();
  });

  it('should fetch and render diagnosis data', async () => {
    renderWithTheme(
      <DiagnosisList
        diagnosisData={diagnosisMockData}
        columnsHeader={diagnosisListColumns}
        currentPage="1"
        lastPage="3"
        prevPage={jest.fn()}
        nextPage={jest.fn()}
        toggleSelectRow={jest.fn()}
        toggleSelectAll={jest.fn()}
        onSelectMode={jest.fn()}
        activeTab={'anality'}
        setActiveTab={jest.fn()}
      />,
    );

    // Espera pela renderização dos dados após a chamada da API
    await screen.findByText(diagnosisMockData[0].animalId); // Verifique se o dado mockado é renderizado

    expect(screen.getByText(diagnosisMockData[0].animalId)).toBeInTheDocument();
  });
});
