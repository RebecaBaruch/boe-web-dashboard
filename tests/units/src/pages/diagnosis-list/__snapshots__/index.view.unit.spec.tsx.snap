// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiagnosisList should render DiagnosisList with text DiagnosisList 1`] = `
<div
  style="isolation: isolate;"
>
  <h1>
    DiagnosisList - Generated page
  </h1>
</div>
`;
